docs/4.4/api-reference.md

```diff
@@ -796,14 +798,129 @@ if err := client.DeleteAccessRequest(ctx, accessReqID); err != nil {
+
+### Certificate Rotation
+
+To maintain security across your cluster, it is a good idea to set up automatic [certificate rotation](architecture/authentication.md#certificate-rotation).
+
+You can use Teleport's `auto` rotation mode to rotate the CA with a default or custom schedule, or you can use `manual` mode to create a custom automated solution that manually triggers each phase.
+
+A carefully implemented `manual` mode solution has the potential to be more fault tolerant and faster, due to the arbitrary nature of rotation schedules and grace periods (explained below).
+
+**Rotation Phases**
+
+A certificate rotation occurs in a series of phases, either triggered automatically or manually.
+
+1. `Standby`: No rotation operations underway. This is the beginning and end state of every CA rotation. If a CA's rotation is not in the standby state, a new rotation cannot begin.
+2. `Init`: New Certificate Authority is issued, but it remains unused while users and servers get updated certificates.
+3. `Update Clients`: Client credentials will have to be updated and reloaded, but servers will still use and respond with old credentials (grace period).
+4. `Update Servers`: Servers will have to reload and should start serving TLS and SSH certificates signed by new CA (retrieved in previous phase).
+5. `Rollback`: Rollback moves back both clients and servers to use the old credentials, but will continue to trust new credentials as well. Must be triggered Manually.
+
+The phases must occur in the order `Init -> Update Clients -> Update Servers` with the beginning and ending resting state being `Standby`. `Rollback` can occur after any phase.
+
+**Automated Rotation**
+
+You can use `tctl auth rotate --type=user --grace-period=10h` or the following RPC to start the rotation in `auto` mode.
 
 ```go
-ca, err := client.GetCertAuthority(services.CertAuthID{
-  DomainName: clusterName,
-  Type:       services.HostCA,
-}, false)
-if err != nil {
+// This will start an automatic rotation that schedules each phase of the rotation in equal increments (each 1/3 of the grace period).
+gracePeriod := time.Hour * 24
+req := auth.RotateRequest{
+  Mode: services.RotationModeAuto,
+  GracePeriod: &gracePeriod, // defaults to 48 hours
+  Type: services.UserCA, // Leave empty to target UserCA and HostCA
+}
+if err := client.RotateCertAuthority(req); err != nil {
+  return err
+}
+```
+
+The grace period should be set to 2-3 times the expected time to rotate the CA to ensure completion, while minimizing the grace period. The `type` flag is useful if you want to rotate the user and host CA's with different strategies or frequencies.
+
+**Automated Rotation with a Custom Schedule**
+
+You can also rotate certificates with a custom schedule. You may want to extend the length of one phase in the schedule without having a ridiculously long grace period, since long grace periods present a possible vulnerability.
```

Comment:
Using a custom certificate rotation schedule will allow you to target specific phase(s) and extend their length without having to use an unnecessarily long grace period, ...